Switch shell to using ToolbarHandler

PureWeen opened the pull request:
### Description of Change

- Shift shells toolbar to using `ToolbarHandler` this way all the logic for managing features of the `Toolbar` isn't duplicated
- Set the `Icons` for the WinUI `NavigationView` to a default color so that they display

### Issues Fixed

Fixes #4506
Fixed #4774

PureWeen committed:
Switch shell to using ToolbarHandler

rachelkang reviewed (commented):
Is the font icon issue also fixed for iOS?

PureWeen committed:
- update accessibility after applying changes

PureWeen commented on src/Controls/src/Core/Platform/Android/Extensions/ToolbarExtensions.cs:
```diff
@@ -100,7 +105,13 @@ public static void UpdateBackButton(this AToolbar nativeToolbar, Toolbar toolbar
+				}
+				else
+				{
+					nativeToolbar.SetNavigationContentDescription(Resource.String.nav_app_bar_open_drawer_description);
```
Can you create an issue for this?  The `Toolbar` hasn't caught up yet to all the features of shell. We'll need to add a property to `Toolbar.cs` so that this is settable from the extension methods. `ShellToolbarTracker` will set these values based on user settings when using `Shell` but we'll need to expose something for non-shell apps

rachelkang replied:
https://github.com/dotnet/maui/issues/5762

PureWeen commented:
> Is the font icon issue also fixed for iOS?

I main against shell and the xbox font icon showed up for me. So AFAICT it was fixed when we moved shell over from renderers to handlers

Blinchik91 commented:
issue #4774 still exists... font icons not showing on iOS, can someone help ?
